baseapp/baseapp.go

```diff
@@ -1060,6 +1080,48 @@ func (app *BaseApp) runMsgs(ctx sdk.Context, msgs []sdk.Msg, msgsV2 []protov2.Me
 	}, nil
 }
 
+// simulateNestedMessages simulates a message nested messages.
+func (app *BaseApp) simulateNestedMessages(ctx sdk.Context, msg HasNestedMsgs) error {
+	msgs, err := msg.GetMsgs()
+	if err != nil {
+		return err
+	}
+
+	if err := validateBasicTxMsgs(app.msgServiceRouter, msgs); err != nil {
+		return err
+	}
+
+	msgsV2, err := app.msgsV1ToMsgsV2(msgs)
+	if err != nil {
+		return err
+	}
+
+	_, err = app.runMsgs(ctx, msgs, msgsV2, execModeSimulate)
```

Comment:
Can a nested message have nested messages, too?

Reply:
I just updated to check also those but I had to use unpack to get msgsV2. Otherwise, it panics when getting the signer of the nested messages with nested messages.